src/salaryAndBenefits/utils/calculateSalary.ts

```diff
@@ -0,0 +1,29 @@
+import { payscale } from "./salaryData";
+  
+interface PayScale {
+    [year: string]: {
+      [examinationYear: string]: string;
+    };
+  }
+  
+  const salaryPayscale: PayScale = payscale;
+  
+  export function calculateSalary(
+    currentYear: number,
+    examinationYear: number,
+    degree: string
+  ): string {
+
+    if (examinationYear >= currentYear) {
+        //TODO: add error handling for data that is not in the payscale
+        console.log("Examination year cannot be in the future.");
+    }      
+
+    const adjustedYear = (examinationYear + (degree === "Bachelor" ? 1 : 0));
```

Comment:
hmm could we simplify this for readability? 
```suggestion
const constValue = degree === "Bachelor" ? 1 : 0. // give this a proper name
    const adjustedYear = (examinationYear + constValue;
```

Reply:
Not convinced this is more readable, at least not without a very good name